internal/setup/setup_test.go

```diff
@@ -1648,6 +1649,45 @@ func runParseReleaseTests(c *C, tests []setupTest) {
 	}
 }
 
+func (s *S) TestPackageMarshalYAML(c *C) {
+	for _, test := range setupTests {
+		c.Logf("Summary: %s", test.summary)
+
+		if test.relerror == "" || test.release == nil {
+			continue
+		}
+
+		data, ok := test.input["chisel.yaml"]
+		if !ok {
+			data = string(defaultChiselYaml)
+		}
+
+		dir := c.MkDir()
+		// Write chisel.yaml.
+		fpath := filepath.Join(dir, "chisel.yaml")
+		err := os.MkdirAll(filepath.Dir(fpath), 0755)
+		c.Assert(err, IsNil)
+		err = os.WriteFile(fpath, testutil.Reindent(data), 0644)
+		c.Assert(err, IsNil)
+		// Write the packages YAML.
+		for _, pkg := range test.release.Packages {
+			fpath = filepath.Join(dir, pkg.Path)
+			err = os.MkdirAll(filepath.Dir(fpath), 0755)
+			c.Assert(err, IsNil)
+			pkgData, err := yaml.Marshal(pkg)
+			c.Assert(err, IsNil)
+			err = os.WriteFile(fpath, testutil.Reindent(string(pkgData)), 0644)
+			c.Assert(err, IsNil)
+		}
+
+		release, err := setup.ReadRelease(dir)
+		c.Assert(err, IsNil)
+
+		release.Path = ""
+		c.Assert(release, DeepEquals, test.release)
```

Comment:
This is a good verification that we're managing to get out what we put in, but it's not enough for our purposes, when a lot of the content that is being committed in this PR is about the actual presentation of the yaml. Where are all those decisions being tested so we get some certainty that we won't rollback a proper display?

This matters because this content is being generated for human eyes, not for a machine.

Reply:
We are testing the actual representation in `cmd_info_test.go`. The rationale was that there are two different modules with different expectations, in this case `setup` and `cmd_info`. `setup` is more abstract and it only cares about the data being correctly marshaled and unmarshaled, that is its promise to consumers. It is `cmd_info` that cares about the representation of the data (e.g. the flow attribute), hence why we put the tests there. We may have several representations in the future and `cmd_info` is only one of them, which is why the tests are there and the promise is not being made at the `setup` level.

With that said, I do not have a strong opinion either way and I understand the reasoning for moving the tests here, so we can do that as well.

Reply:
The command tests should test the command and make sure it works, including the actual generation of an output because we need to ensure things are properly wired together.  With that said, it has zero clue about how rich the representation, is and what concerns it has, what data types exist, and so forth. The code that does all of that is here, and you don't want to be editing code here and then adding a test there for every one of those changes.  For that reason, I bet the code here is under-tested, because we didn't really pay attention to the details of what we were writing and backing these with tests.

Reply:
Added `TestPackageYAMLFormat` in setup_test to check for proper format. Thank you.